image_services/opera/update_opera_uris.py

```diff
@@ -0,0 +1,76 @@
+import argparse
+import csv
+import json
+import logging
+import os
+from pathlib import Path
+
+import boto3
+import requests
+
+S3_CLIENT = boto3.client('s3')
+log = logging.getLogger(__name__)
+
+
+def query_cmr(polarization):
+    session = requests.Session()
+    search_url = 'https://cmr.earthdata.nasa.gov/search/granules.umm_json'
+
+    params = {
+        'short_name': 'OPERA_L2_RTC-S1_V1',
+        'attribute[]': f'string,POLARIZATION,{polarization}',
+        'page_size': 2000,
+    }
+    headers = {}
+    vsis3_uris = []
+    while True:
+        response = session.get(search_url, params=params, headers=headers)
+        response.raise_for_status()
+        for granule in response.json()['items']:
+            for url in granule['umm']['RelatedUrls']:
+                if url['URL'].startswith('s3://') and url['URL'].endswith(f'{polarization}.tif'):
+                    vsis3_uris.append(url['URL'].replace('s3://', '/vsis3/'))
+                    break
+        if 'CMR-Search-After' not in response.headers:
+            break
+        headers['CMR-Search-After'] = response.headers['CMR-Search-After']
+        return vsis3_uris
+
+
+def upload_file_to_s3(path_to_file: Path, bucket: str, prefix: str = ''):
+    path_to_file = Path(path_to_file)
```

Comment:
It feels a little wrong to ask for a Path and then cast to Path right away. Maybe ask for a string and cast to Path, or cast to Path before you use the function.